openapi-v2.yaml

```diff
@@ -671,4 +732,115 @@ paths:
         default:
           description: error response
           schema:
-            $ref: "#/definitions/Error"
\ No newline at end of file
+            $ref: "#/definitions/Error"
+
+  /credentials/{namespace}:
+    parameters:
+      - name: namespace
+        in: path
+        description: namespace
+        required: true
+        type: string
+      - name: provider
+        in: query
+        description: "Show only the credentials from this provider"
+        type: string
+        enum: *CLOUDPROVIDER
```

Comment:
Is this syntax valid? Or should it be `$ref: "#/definitions/CloudProvider"`?

Reply:
This is a piece of OpenAPI v2 weirdness. OpenAPI v2 [doesn’t support `$ref` in this specific case](https://swagger.io/docs/specification/2-0/enums/) (cf. the “Reusable Enums” section), so you have use a [native YAML reference](https://yaml.org/spec/1.2/spec.html#id2765878). (Annoyingly, it *does* support referencing enums with `$ref` in *other* cases, so it took way too long for me to understand what was going on when I tried using `$ref` here.)

(unrelated complaint: even apart from this specific issue, i do not like yaml.)

Reply:
Let's make sure that this will work with our tools (client generators, bundlers etc), because i got an error while trying to build & run that. I had to delete that enum line and leave just the `string` like we do in other similar cases. Not the best solution but...

Reply:
We use the `ref` style definition with `DATATYPE` and it works "fine" (not sure if its idea but doesn't error and auto generaters), so I think we can handle this here. This does break our autogeneration so we need to find something that works for the `go-swagger` and `openapi tools`.

Reply:
The difference between here and the `Datatype` `$ref` is that the reference to Datatype is in a `properties: type:` block, and other uses of `$ref` are likewise in `schema:` blocks. `$ref` *specifically* does not work here and using it would cause the checks (and generation tools) to fail. See [one run where I was using `$ref` and the mysterious failure it caused](https://dev.azure.com/TileDB-Inc/CI/_build/results?buildId=14028&view=logs&j=f164dbd1-368a-58ba-3795-906e0bb83435&t=34e91703-cf6b-5e24-170c-f8135c3c10c4).

Since apparently including the YAML reference breaks things, I’ll follow up with a change to remove it. The `enum` list is just a hint to callers, so it can be removed with no harm.

Reply:
> specifically does not work here

Missed that it was the location that was the issue, thanks for clarifying!